snowflake/src/integration/java/org/apache/iceberg/snowflake/NamespaceTests.java

```diff
@@ -0,0 +1,290 @@
+    }
+  }
+
+  @Test
+  public void testLoadNonExistingRootLevelNamespace() {
+    String nonExistingDb = "IDontExist";
+    Assertions.assertThatThrownBy(
+            () -> {
+              snowflakeCatalog.loadNamespaceMetadata(Namespace.of(nonExistingDb));
+            })
+        .isInstanceOf(NoSuchNamespaceException.class);
+  }
+
+  @Test
+  public void testLoadNonExistingDBLevelNamespace() {
+    String nonExistingSchema = "IDontExist";
+    String db = TestConfigurations.getInstance().getDatabase();
+    Assertions.assertThatThrownBy(
+            () -> {
+              snowflakeCatalog.loadNamespaceMetadata(Namespace.of(db, nonExistingSchema));
+            })
+        .isInstanceOf(NoSuchNamespaceException.class);
+  }
+
+  @Test
+  public void testLoadNamespaceThatExceedMaxSupportedHierarchy() {
+    String nonExistingSchema = "IDontExist";
+    String db = TestConfigurations.getInstance().getDatabase();
+    Assertions.assertThatThrownBy(
+            () -> {
+              snowflakeCatalog.loadNamespaceMetadata(
+                  Namespace.of(db, nonExistingSchema, "UnSupportedLevel"));
+            })
+        .isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void testLoadNamespaceWithUnquotedIdentifier() throws SQLException, InterruptedException {
+    String schema1 = "Schema123";
+    String dbName = TestConfigurations.getInstance().getDatabase();
+    try {
+      Assertions.assertThatNoException()
+          .isThrownBy(
+              () -> {
+                clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+                createOrReplaceSchema(schema1);
+                snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, schema1));
+              });
+    } finally {
+      dropSchemaIfExists(schema1);
+    }
+  }
+
+  @Test
+  public void testLoadNamespaceWithUnquotedIdentifierIsCaseInsensitive()
+      throws SQLException, InterruptedException {
+    String schema1 = "Schema123";
+    String dbName = TestConfigurations.getInstance().getDatabase();
+    try {
+      Assertions.assertThatNoException()
+          .isThrownBy(
+              () -> {
+                clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+                createOrReplaceSchema(schema1);
+                snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, schema1));
+                snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, schema1.toUpperCase()));
+                snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, "schEmA123"));
+              });
+    } finally {
+      dropSchemaIfExists(schema1);
+    }
+  }
+
+  @Test
+  public void testLoadNamespaceWithUnquotedIdentifierWithLeadingUnderscoreAndDollarCharacters()
+      throws SQLException, InterruptedException {
+    String schema1 = "_Schema$_123";
+    String dbName = TestConfigurations.getInstance().getDatabase();
+    try {
+      Assertions.assertThatNoException()
```

Comment:
same as above

Reply:
Switched to parameterized tests and it simplified tests. Thanks for the suggestion!